src/bitnodes.cpp

```diff
@@ -0,0 +1,223 @@
+        {
+            if (!error){
+                std::istream response_stream(&response_);
+                std::string http_version;
+                unsigned int status_code;
+                response_stream >> http_version;
+                response_stream >> status_code;
+                //std::string status_message;
+                //std::getline(response_stream, status_message);
+                if (!response_stream || http_version.substr(0, 5) != "HTTP/")
+                {
+                    throw runtime_error("Bitnodes response not HTTP");
+                }
+                if (status_code != 200)
+                {
+                    throw runtime_error(strprintf("Bitnodes returned HTTP status %d\n", status_code));
+                }
+
+                // Skip all headers
+                string header;
+                while (std::getline(response_stream, header) && header != "\r");
+
+                // Read until EOF
+                boost::asio::async_read(socket_, response_,
+                        boost::asio::transfer_at_least(1),
+                        boost::bind(&client::handle_read, this, boost::asio::placeholders::error, boost::asio::placeholders::bytes_transferred));
+            }
+            else {
+                throw runtime_error(strprintf("Bitnodes HTTP read headers error: %s\n", error.message().c_str()));
+            }
+        }
+
+
+        void handle_read(const boost::system::error_code& error, size_t bytes_transferred)
+        {
+            if (!error){
+                std::ostringstream ss ;
+                ss << &response_;
+                content_.append(ss.str());
+
+                // Read until EOF.
+                boost::asio::async_read(socket_, response_,
+                        boost::asio::transfer_at_least(1),
+                        boost::bind(&client::handle_read, this, boost::asio::placeholders::error, boost::asio::placeholders::bytes_transferred));
+            } else {
+                timer_.get()->cancel();
+                if (error != boost::asio::error::eof) {
+                    throw runtime_error(strprintf("Bitnodes HTTP read error: %s\n", error.message().c_str()));
+                }
+            }
+        }
+
+        std::string getContent() const {
+            return content_;
+        }
+
+    private:
+        boost::asio::ssl::stream<boost::asio::ip::tcp::socket> socket_;
+        boost::asio::streambuf response_;
+        std::string content_;
+        boost::scoped_ptr<boost::asio::deadline_timer> timer_;
+        std::string url_path_;
+};
+
+
+/**
+ * Pass in an empty string vector to be populated with ip:port strings
+ */
+bool GetLeaderboardFromBitnodes(vector<string>& vIPs)
+{
+    // Bitnodes connection parameters
+    string url_host = "bitnodes.21.co";
+    string url_port = "443";
+    string url_path = "/api/v1/nodes/leaderboard/?limit=100";
+    string cert_hostname = "dazzlepod.com";
```

Comment:
It's the hostname on the certificate.  I spoke with Bitnodes creator @ayeowch about this a while ago:

> ...that is how the PTR record has always been configured and will remain the same as long as the project is up.  So yes, you can use dazzlepod.com for hostname check.